extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/B3PropagatorExtractorMultipleHeaders.java

```diff
@@ -28,33 +28,38 @@
       Context context, C carrier, TextMapPropagator.Getter<C> getter) {
     Objects.requireNonNull(carrier, "carrier");
     Objects.requireNonNull(getter, "getter");
-    SpanContext spanContext = getSpanContextFromMultipleHeaders(carrier, getter);
-    if (!spanContext.isValid()) {
-      return Optional.empty();
-    }
-
-    return Optional.of(context.with(Span.wrap(spanContext)));
+    return extractSpanContextFromMultipleHeaders(context, carrier, getter);
   }
 
-  private static <C> SpanContext getSpanContextFromMultipleHeaders(
-      C carrier, TextMapPropagator.Getter<C> getter) {
+  private static <C> Optional<Context> extractSpanContextFromMultipleHeaders(
+      Context context, C carrier, TextMapPropagator.Getter<C> getter) {
     String traceId = getter.get(carrier, TRACE_ID_HEADER);
     if (StringUtils.isNullOrEmpty(traceId)) {
-      return SpanContext.getInvalid();
+      return Optional.empty();
     }
     if (!Common.isTraceIdValid(traceId)) {
       logger.fine(
           "Invalid TraceId in B3 header: " + traceId + "'. Returning INVALID span context.");
-      return SpanContext.getInvalid();
+      return Optional.empty();
     }
 
     String spanId = getter.get(carrier, SPAN_ID_HEADER);
     if (!Common.isSpanIdValid(spanId)) {
       logger.fine("Invalid SpanId in B3 header: " + spanId + "'. Returning INVALID span context.");
-      return SpanContext.getInvalid();
+      return Optional.empty();
+    }
+
+    // if debug flag is set, then set sampled flag, and also store the B3 debug flag in the context
+    // for onward use by B3 injector
+    String debug = getter.get(carrier, DEBUG_HEADER);
+    if (!StringUtils.isNullOrEmpty(debug) && debug.contentEquals(Common.TRUE_INT)) {
```

Comment:
```suggestion
    if (Common.TRUE_INT.equals(debug)) {
```

throughout PR. Empty check is redundant, and it's always `String`, not `CharSequence` so can use `equals`.

Reply:
tidied this up